cmd/src/batch_common.go

```diff
@@ -268,9 +268,13 @@ func executeBatchSpec(ctx context.Context, ui ui.ExecUI, opts executeBatchSpecOp
 		return err
 	}
 
-	// In the past, we checked `docker version`, but now we retrieve the number
-	// of CPUs, since we need that anyway and it performs the same check (is
-	// Docker working _at all_?).
+	// In the past, we relied on `getBatchParallelism` to ascertain if docker is running,
+	// however, we don't always check for the number of CPUs (especially when the -j parallelis)
+	// flag is passed. This is a more explicit check to confirm docker is working.
+	if err := checkExecutable("docker", "version"); err != nil {
+		return err
+	}
```

Comment:
I think this makes src-cli prone to the same "hanging" issue again that we were running into before, `getBatchParallelism` uses `withFastCommandContext(ctx)` to get around that. @LawnGnome any thoughts?

Reply:
Oh okay. 

Would a better alternative be to create a method that still runs `docker version` but with the `withFastCommandContext(ctx)` ? WDTY

Reply:
Yes I think so!

Reply:
Yep, that would be the best thing to do. Sorry I missed that yesterday!

Reply:
No problem, that's what reviews are for - to catch this kind of things.
I've implemented the check using `withFastCommandContext(ctx)` though.